tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -298,14 +299,9 @@ internal ImmutableTracerSettings(TracerSettings settings, bool unusedParamNotToU
         /// </summary>
         /// <seealso cref="ConfigurationKeys.CustomSamplingRules"/>
         [GeneratePublicApi(PublicApiUsage.ImmutableTracerSettings_CustomSamplingRules_Get)]
-        internal string? CustomSamplingRulesInternal { get; }
+        internal string? CustomSamplingRulesInternal => DynamicSettings.SamplingRules ?? _customSamplingRulesInternal;
 
-        /// <summary>
-        /// Gets the trace sampling rules from remote config.
-        /// These contain custom remote rules and dynamic (aka adaptive) rules.
-        /// They will be merged with local sampling rules.
-        /// </summary>
-        internal string? RemoteSamplingRules => DynamicSettings.SamplingRules;
+        internal bool CustomSamplingRulesInternalIsRemote => DynamicSettings.SamplingRules != null;
```

Comment:
nit: you don't need the `Internal` in here - that's only required when we have auto-public-api-generation (and we can fix all that in v3 🎉)

Reply:
renamed